xdsl/parser.py

```diff
@@ -1,1335 +1,1924 @@
 from __future__ import annotations
 
+import ast
+import contextlib
+import functools
+import itertools
+import re
+import sys
+import traceback
```

Comment:
These are all builtin python modules

Reply:
I am surprised the parser needs the python ast?

Reply:
We use it to evaluate string literals, that is actually quite tricky, and instead of re-inventing the wheel I looked for a stdlib function that handles that for us. That's why I used `ast.literal_eval`. We also could have used `json.loads`, or something else. I just found `literal_eval` first while searching/thinking about it.